### DISCUSSION TITLE
Is it possible to change the workspace when clicking on the spaces made from sketchybar?

### DISCUSSION THREAD

**Discussion by RaiSaugat** [FelixKratz/SketchyBar#490] [Q&A]
![image](https://github.com/FelixKratz/SketchyBar/assets/19557538/2e12abc6-1b90-4ea5-8033-a9a9088067e6)

For example, I have this on my spaces, will I be able to click on 6th space so that the workspace will switch to 6?

**Comment by Pe8er** [comment]
This could help you (pulled from @FelixKratz 's setup). Besides activating a space on click, it also lets you name spaces if you right-click or shift-left-click.

```bash
mouse_clicked() {
  if [ "$BUTTON" = "right" ] || [ "$MODIFIER" = "shift" ]; then
    SPACE_NAME="${NAME#*.}"
    SPACE_LABEL="$(osascript -e "return (text returned of (display dialog \"Rename space $SPACE_NAME to:\" default answer \"\" with title \"Space Renamer\" buttons {\"Cancel\", \"Rename\"} default button \"Rename\"))")"
    if [ $? -eq 0 ]; then
      if [ "$SPACE_LABEL" = "" ]; then
        set_space_label "${NAME:6}"
      else
        set_space_label "${NAME:6} $SPACE_LABEL"
      fi
    fi
  else
    yabai -m space --focus $SID 2>/dev/null
  fi
}
```

Also make sure your item is subscribed to `mouse.clicked` event.

You can see how it's implemented [over here](https://github.com/Pe8er/dotfiles/blob/master/config.symlink/sketchybar/plugins/app_space.sh).

**Reply by RaiSaugat** [reply]
Thanks, I'll sure give it a try.

**Reply by RaiSaugat** [reply]
Seems a lot of code :sweat_smile:, is there any demos or screen recording of how it would look like?
